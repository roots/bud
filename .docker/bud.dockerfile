FROM node:lts

LABEL name 'bud'
LABEL version 1

RUN npm install -g verdaccio pm2

COPY ./.docker/bud/motd /etc/motd
COPY ./.docker/bud/bash.bashrc /etc/bash.bashrc
