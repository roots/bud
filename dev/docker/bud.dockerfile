FROM node:16

LABEL name 'bud'
LABEL version 1

RUN npm install netlify-cli npm-cli-login --global

COPY ./dev/docker/bud/motd /etc/motd
COPY ./dev/docker/bud/bash.bashrc /etc/bash.bashrc

COPY ./ /srv/bud

RUN mkdir -p /srv/bud/node_modules \
  && mkdir -p /srv/bud/storage/verdaccio \
  && mkdir -p /srv/bud/storage/yarn \
  && mkdir -p /srv/bud/storage/coverage \
  && mkdir -p /srv/mocks
