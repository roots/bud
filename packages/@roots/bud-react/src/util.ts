import type {Configuration} from 'webpack'

function addRefresh(entries, [name, assets]) {
  return {
    'react-dev': {
      import: ['react-refresh/runtime'],
    },
    ...(entries ?? {}),
    [name]: {
      ...assets,
      dependOn: ['react-dev', ...(assets.dependOn ?? [])],
    },
  }
}

function reducer(entry: Configuration['entry']) {
  return entry
    ? Object.entries(entry).reduce(addRefresh, entry)
    : {}
}

export default reducer
