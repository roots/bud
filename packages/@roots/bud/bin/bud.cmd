@echo off

node "%~dp0\bud.js" %*
