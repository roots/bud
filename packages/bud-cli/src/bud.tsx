#!/usr/bin/env node
import {yargs} from '@roots/bud-support'
import * as build from './commands/build'
import * as publish from './commands/publish'

/**
 * $bud
 */
const cli = yargs
  .command(build)
  .command(publish)
  .recommendCommands()
  .demandCommand(1, 'Try using one of the above commands.')
  .usage('bud [command] [options]')
  .version()
  .wrap(yargs.terminalWidth())
  .showHelpOnFail(true)
  .epilog('https://github.com/roots/bud').argv

export {cli as bud}
module.exports = {bud: cli}
