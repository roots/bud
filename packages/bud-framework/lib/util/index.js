"use strict";
exports.__esModule = true;
exports.util = void 0;
var dump_1 = require("./dump");
var fab_1 = require("./fab");
var format_1 = require("./format");
var notify_1 = require("./notify");
var projectRoot_1 = require("./projectRoot");
var shortCircuit_1 = require("./shortCircuit");
var terminate_1 = require("./terminate");
var processHandler_1 = require("./processHandler");
var fs_1 = require("./fs");
var os_1 = require("./os");
var usedExt_1 = require("./usedExt");
var logger_1 = require("./logger");
exports.util = {
    format: format_1.format,
    fs: fs_1.fs,
    dump: dump_1.dump,
    logger: logger_1.logger,
    shortCircuit: shortCircuit_1.shortCircuit,
    fab: fab_1.fab,
    notify: notify_1.notify,
    projectRoot: projectRoot_1.projectRoot,
    processHandler: processHandler_1.processHandler,
    terminate: terminate_1.terminate,
    usedExt: usedExt_1.usedExt,
    os: os_1.os
};
//# sourceMappingURL=index.js.map