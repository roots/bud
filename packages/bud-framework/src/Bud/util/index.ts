import {when} from './when'
import {proxy} from './proxy'
import pretty from './pretty'
import format from './format'
import logger from './logger'

export default {pretty, format, logger, when, proxy}
