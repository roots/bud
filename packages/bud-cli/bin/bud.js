#!/usr/bin/node
require('./../lib/bud')
