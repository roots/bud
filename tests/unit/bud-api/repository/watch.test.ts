import {Bud, factory} from '../../../util/bud'

describe('bud.watch', function () {
  let bud: Bud

  beforeAll(async () => {
    bud = await factory({
      mode: 'development',
    })
  })

  it('is a function', () => {
    expect(bud.watch).toBeInstanceOf(Function)
  })

  it("doesn't throw when called in production", () => {
    expect(bud.watch(['**/*.js'])).toBeInstanceOf(Bud)
  })

  it('sets watch files', async () => {
    const files = ['**/*.js']

    bud.watch(files)

    await bud.api.processQueue()

    expect(bud.store.get('server.watch.files')).toMatchSnapshot(
      files,
    )
  })
})
