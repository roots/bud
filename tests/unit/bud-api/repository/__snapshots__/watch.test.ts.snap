// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bud.watch sets watch files 1`] = `
Array [
  "1/*.js",
]
`;

exports[`bud.watch sets watch files 2`] = `
Array [
  "1/*.js",
  "2/*.js",
]
`;
